### PR TITLE
rename mutableUsers to onlyDeclarative

### LINKED ISSUE #9192: mutableUsers is named confusingly
You can't know if "true" means NixOS can mutate users, or if you can do it yourself.
Something like users.declarative would be more clear I think.

@cillianderoiste also said he found the name confusing.

### PR DESCRIPTION
**Pull request by magnetophon**
###### Motivation for this change

Fix #9192.
###### Things done

NOT TESTED!

I have 0 experience with VM's and I probably don't even have enough RAM.
Could someone test this?
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: nixos/modules/programs/shadow.nix [hunk 1]
```diff
@@ -103,7 +103,7 @@ in
       };
 
     security.setuidPrograms = [ "su" "chfn" ]
-      ++ lib.optionals config.users.mutableUsers
+      ++ lib.optionals config.users.onlyDeclarative
```
Comment (rasendubi) [inline nixos/modules/programs/shadow.nix]: Changed behaviour

File: nixos/modules/programs/shadow.nix [hunk 2]
```diff
@@ -60,7 +60,7 @@ in
   config = {
 
     environment.systemPackages =
-      lib.optional config.users.mutableUsers pkgs.shadow ++
+      lib.optional config.users.onlyDeclarative pkgs.shadow ++
```
Comment (rasendubi) [inline nixos/modules/programs/shadow.nix]: Changed behaviour

Comment (magnetophon) [inline nixos/modules/programs/shadow.nix]: @rasendubi 
I'm sorry, but idk how to fix these.
Do I just put a ``!`` in front of ``config.users.onlyDeclarative``?

Comment (rasendubi) [inline nixos/modules/programs/shadow.nix]: Yes. Parentheses might also be needed.

**Comment by magnetophon** [comment]
@rasendubi Sorry, I tried to rebase but made a second commit instead.
I'll rebase again when there are no other changes.

**Comment by magnetophon** [comment]
Pinging the interested people from #9192:
@pSub, @bjornfor, @lethalman, @wmertens.

**Comment by bjornfor** [comment]
IMHO, the old option name should still work, but give a warning and point to the new name. (Is that what rename.nix does?) As of now, it looks like an incompatible change.

**Comment by domenkozar** [comment]
We can't just rename since the new value is negative. We have to handle
that otherwise we can't land this change.

**Comment by magnetophon** [comment]
@bjornfor @domenkozar Well, this is as far as I can take it, with my limited knowledge.
I'll need some guidance from here on.

**Comment by joachifm** [comment]
I'm :-1:  on this, I don't think the old option is confusing at all, so to me this is just a nuisance waiting to happen.

Anyway, I think instead of renaming, you should at least add a `mkRemovedOptionModule` for `mutableUsers`. I don't see a good way of communicating what the correct uprade is.

**Comment by magnetophon** [comment]
I'll close this for now.
If someone wants to finish it, I'll be happy though!

**Pull request closed without merging**